### DISCUSSION TITLE
Creating Profiles

### DISCUSSION THREAD

**Discussion by rusty-snake** [netblue30/firejail#4444] [Wiki]
Discussion for discussions about https://github.com/netblue30/firejail/wiki/Creating-Profiles.

Predecessor: #2748

**Comment by Markus-N** [comment]
It would be helpful to add a list of all available variables, maybe including a little bit of explanation for some of them.  
E.g. what does ``${PATH}`` mean? Are ``${DOCUMENTS}``, ``${MUSIC}``, ... language dependent or fixed?

**Reply by rusty-snake** [reply]
In case anyone wants to formulate this:

- `$HOME`: Blocked to catch mistakes.
- `${HOME}`/`~`: self-explanatory.
- `${CFG}`: `SYSCONFDIR`, usually `/etc/firejail` (or `/usr/local/etc/firejail`).
- `${RUNUSER}`: `/run/user/$UID` (For `$XDG_RUNTIME_DIR` support see #4535)
- `${PATH}/foo`: all occurrences of `foo` in `$PATH`-directories
- `${DOWNLOADS}`, `${MUSIC}`, `${VIDEOS}`, `${PICTURES}`, `${DESKTOP}`, `${DOCUMENTS}`: Resolved from `~/.config/user-dirs.dirs` (not working for paths outside of `$HOME`, #3735) with hardcoded fallback for the default `English, Russian, French, Italian, Spanish, Portuguese, German` names.

**Reply by Markus-N** [reply]
Oh whow ... that was quick.  
One more question just came into my mind: Are symlinks dereferenced ? E.g. I have /home/Videos symlinked to another filesystem.

**Reply by rusty-snake** [reply]
Try it out ;)
It will depend on the command, for `blacklist` yes, for `whitelist` things get complicated if the target is outside of `$HOME`.

**Reply by Markus-N** [reply]
OK, thanx.
